lib/slide_hub/convert_procedure.rb

```diff
@@ -0,0 +1,113 @@
+require 'batch_logger'
+require 'cloud/queue/response'
+require 'convert_util'
+require 'image'
+require 'tmpdir'
+
+class ConvertProcedure
+  def initialize(object_key:, logger:)
+    @object_key = object_key
+    @logger = logger
+    @work_dir = ''
+    @work_file = SecureRandom.hex.to_s
+    @file_type = ''
+    @slide_image_list = []
+    @upload_file_list = []
+  end
+
+  def run
+    Dir.mktmpdir do |dir|
+      @work_dir = dir
+      logger.info("Current directory is #{work_dir}")
+      return false unless save_file
+      convert_to_ppm
+      convert_to_jpg
+      generate_json
+      convert_to_thumbnail
+      convert_to_transcript
+      upload_files
+      update_database
+      true
+    end
+  end
+
+  private
+
+    attr_reader :logger
+    attr_reader :work_dir
+    attr_reader :object_key
+    attr_reader :work_file
+
+    def save_file
+      return false unless CloudConfig::SERVICE.save_file(CloudConfig::SERVICE.config.bucket_name, object_key, "#{work_dir}/#{work_file}")
+      @file_type = SlideHub::ConvertUtil.new.get_slide_file_type("#{work_dir}/#{work_file}")
+      true
+    end
+
+    def convert_to_ppm
```

Comment:
__[RuboCop]__

Metrics/AbcSize: Assignment Branch Condition size for convert_to_ppm is too high. [38.05/30]


<!-- Comment by [SideCI](https://sideci.com) -->